src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -2031,6 +2033,8 @@ void grpc_chttp2_cancel_stream(grpc_chttp2_transport* t, grpc_chttp2_stream* s,
     s->seen_error = true;
   }
   grpc_chttp2_mark_stream_closed(t, s, 1, 1, due_to_error);
+  // Make sure any pending write callbacks are flushed
+  GRPC_CLOSURE_LIST_SCHED(&s->run_after_write);
```

Comment:
same here. remove_stream should run everything

Reply:
removed.